cookbooks/bcpc-hadoop/templates/default/hdp_yarn-site.xml.erb

```diff
@@ -10,97 +10,117 @@
-  </property>
-
-  <% @rm_hosts.each do |h| %>   
-  <property>
-    <name>yarn.resourcemanager.hostname.rm<%= "#{node.chef_environment}#{h[:node_number]}" %></name>
-    <value><%=float_host(h[:hostname]) %></value>
-  </property>
+    <property>
+      <name>yarn.resourcemanager.cluster-id</name>
+      <value><%="#{node.chef_environment}"%></value>
+    </property>
+
+    <property> 
+      <name>yarn.resourcemanager.ha.enabled</name> 
+      <value>true</value>
+    </property> 
+
+    <property>
+      <name>yarn.resourcemanager.ha.rm-ids</name>
+      <value><%=@rm_hosts.map{ |s| "rm#{node.chef_environment}#{s[:node_number]}" }.join(",") %></value>
+    </property>
+
+    <% @rm_hosts.each do |h| %>   
+      <property>
+        <name>yarn.resourcemanager.hostname.rm<%= "#{node.chef_environment}#{h[:node_number]}" %></name>
+        <value><%=float_host(h[:hostname]) %></value>
+      </property>
+    <% end %>
+
+    <% @rm_hosts.each do |h| %>   
+      <property>
+        <name>yarn.resourcemanager.address.rm<%= "#{node.chef_environment}#{h[:node_number]}" %></name>
+        <value><%=float_host(h[:hostname])%>:<%=node["bcpc"]["hadoop"]["yarn"]["resourcemanager"]["port"]%></value>
+      </property>
+    <% end %>
+
+    <property> 
+      <name>yarn.resourcemanager.recovery.enabled</name> 
+      <value>true</value> 
+    </property> 
+ 
+    <property> 
+      <name>yarn.resourcemanager.store.class</name> 
+      <value>org.apache.hadoop.yarn.server.resourcemanager.recovery.ZKRMStateStore</value> 
+    </property> 
+   
+    <property> 
+      <name>yarn.resourcemanager.zk-address</name>
+      <value><%= @zk_hosts.map{ |s| float_host(s[:hostname]) + ":#{node[:bcpc][:hadoop][:zookeeper][:port]}" }.join(",") %></value>
+    </property>
+   
+    <property> 
+      <name>yarn.client.failover-proxy-provider</name> 
+      <value>org.apache.hadoop.yarn.client.ConfiguredRMFailoverProxyProvider</value> 
+    </property>
+    <% else %>
+
+    <property>
+      <name>yarn.resourcemanager.resource-tracker.address</name>
+      <value><%="#{float_host(@rm_hosts.first[:hostname])}:8031" if not @rm_hosts.empty?%></value>
+    </property>
+
+    <property>
+      <name>yarn.resourcemanager.address</name>
+      <value><%="#{float_host(@rm_hosts.first[:hostname])}:#{node["bcpc"]["hadoop"]["yarn"]["resourcemanager"]["port"]}" if not @rm_hosts.empty?%></value>
+    </property>
+
+    <property>
+      <name>yarn.resourcemanager.scheduler.address</name>
+      <value><%="#{float_host(@rm_hosts.first[:hostname])}:8030" if not @rm_hosts.empty?%></value>
+    </property>
+
+    <property>
+      <name>yarn.resourcemanager.admin.address</name>
+      <value><%="#{float_host(@rm_hosts.first[:hostname])}:8033" if not @rm_hosts.empty?%></value>
+    </property>
+
+    <property>
+      <name>yarn.resourcemanager.webapp.address</name>
+      <value><%="#{float_host(@rm_hosts.first[:hostname])}:8088" if not @rm_hosts.empty?%></value>
```

Comment:
Again, this is still a YARN default but not modified here, only indentation changed. At some point, a more thorough rationalization would be good (particularly for webapp ports to go through HAProxy for user ease with a single VIP-based URL would be nice).